aspnetcore/blazor/performance/profiling.md

```diff
@@ -0,0 +1,226 @@
+* Web-based testing
+  * Use the JavaScript API to obtain a [promise](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise) of NetTrace (`.nettrace`) bytes.
+  * Upload the file via HTTP.
+  * Parse and validate that the trace contains the expected classes.
+
+## Counters trace
+
+* Manual testing
+  * Browser developer tools: Download the `.json` output file, open the file in Visual Studio, and find the expected counters.
+  * [`dotnet-counters collect`](/dotnet/core/diagnostics/dotnet-counters): Open the `.csv`/`.json` output file in Visual Studio and find the expected counters.
+* Web-based testing
+  * Use the JavaScript API to obtain a [promise](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise) of NetTrace (`.nettrace`) bytes.
+  * Upload the file via HTTP.
+  * Parse and validate that the trace contains the expected counters.
+
+## .NET Core Diagnostics Client Library example
+
+Parse and validate NetTrace (`.nettrace`) messages using the .NET Core Diagnostics Client Library:
+
+* [`dotnet/diagnostics` GitHub repository](https://github.com/dotnet/diagnostics)
+* [`Microsoft.Diagnostics.NETCore.Client` NuGet package](https://www.nuget.org/packages/Microsoft.Diagnostics.NETCore.Client)
+
+For more information, see the [.NET Core diagnostics documentation](/dotnet/core/diagnostics/) and the [`IpcMessage` API (reference source)](https://github.com/dotnet/diagnostics/blob/main/src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcMessage.cs).
+
+[!INCLUDE[](~/includes/aspnetcore-repo-ref-source-links.md)]
+
+The following example:
+
+* Collects a GC (Garbage Collector) dump of the live .NET process.
+* Collects performance counters for 60 seconds.
+* Collects CPU counters for 60 seconds.
+
+In the project file (`.csproj`), the properties in the following table enable integration with the browser's profiler.
+
+Property | Default | Set value to&hellip; | Description
+--- | :---: | :---: | ---
+`<WasmProfilers>` | No value | `browser` | Mono profilers to use. Potential values are "`browser`" and "`log`". To use both, separate the values with a semicolon. The `browser` profiler enables integration with the browser's developer tools profiler.
+`<WasmPerfTracing>` | `false` | `true` | Controls diagnostic server tracing.
+`<WasmPerfInstrumentation>` | `false` | `true` | Controls CPU sampling instrumentation for diagnostic server.
+`<MetricsSupport>` | `false` | `true` | Controls `System.Diagnostics.Metrics` support. For more information, see the [`System.Diagnostics.Metrics` namespace](/dotnet/api/system.diagnostics.metrics).
+`<EventSourceSupport>` | `false`| `true` | Controls `EventPipe` support. For more information, see [Diagnostics and instrumentation: Observability and telemetry](/dotnet/core/deploying/native-aot/diagnostics#observability-and-telemetry).
+
+```xml
+<PropertyGroup>
+  <WasmProfilers>browser</WasmProfilers>
```

Comment:
`<WasmProfilers>browser</WasmProfilers>` is not compatible with `<WasmPerfTracing>true</WasmPerfTracing>` at all. They are separate approach to profiling. When talking about ETW/EventPipe/DiagnostiServer the `WasmPerfTracing` is way to go.